### ISSUE TITLE
[Kicker plugin] Automation connection of a particular knob is not restored when reopening a project

### ISSUE THREAD

**Issue by M4rotte** [LMMS/lmms#6797]
Hi,

I automated multiple knobs of a Kicker plugin, using some automation tracks in the song. I noticed that the connection of the "End distortion" is systematically not restored after closing/reopening the project. This never happen with another automated knob of this plugin, but with "End distortion" it happen every times.

Version used : LMMS 1.3.0-alpha.1.315+g57e7fdc53 (Linux x86_64, Qt 5.15.8, GCC 12.2.0)

**Comment by zonkmachine** [comment]
I can verify this. It is my understanding that this is a problem on loading. I believe it could have been introduced in https://github.com/LMMS/lmms/pull/727 available in [1.1.0-RC4](https://github.com/LMMS/lmms/releases/tag/v1.0.93). I couldn't test that unfortunately since I would need an older platform for this.

Here is the code that fails: https://github.com/LMMS/lmms/blob/bc99728534c90fba25c5deedef1568dc5d9d6e62/plugins/Kicker/Kicker.cpp#L117-L124 I don't know why though. It's like "distend" is invisible to the system. When I put in some code to test for all of the attributes, this is the result. Looks like some namespace voodoo to me. "distend" is clearly present in the resulting project save.

```
Has attribute "version"
Has attribute "startfreq"
Has attribute "endfreq"
Has attribute "decay"
Has attribute "dist"
Hasn't got attribute "distend"
Has attribute "gain"
Has attribute "env"
Has attribute "noise"
Has attribute "click"
Has attribute "slope"
Has attribute "startnote"
Has attribute "endnote"
```
If I instead test for the version of the plugin the loading succeeds (Quickfix below). The version number is 1 and this is defined in **Kicker.h**. It's not literally the same thing as testing for "distend" but pretty much. Version number 1 predates "distend". I was thinking possibly moving this code to **DataFile.cpp** and do a proper upgrade but that's not possible if **hasAttribute("distend")** doesn't work.

Quickfix:
```cpp
diff --git a/plugins/Kicker/Kicker.cpp b/plugins/Kicker/Kicker.cpp
index 7d4c6fbb0..658479193 100644
--- a/plugins/Kicker/Kicker.cpp
+++ b/plugins/Kicker/Kicker.cpp
@@ -114,14 +114,19 @@ void KickerInstrument::loadSettings( const QDomElement & _this )
        m_endFreqModel.loadSettings( _this, "endfreq" );
        m_decayModel.loadSettings( _this, "decay" );
        m_distModel.loadSettings( _this, "dist" );
-       if( _this.hasAttribute( "distend" ) )
+       if( m_versionModel.value() > 0 )
        {
                m_distEndModel.loadSettings( _this, "distend" );
        }
```

**Comment by DomClark** [comment]
Should be fixed by #6670.

**Comment by zonkmachine** [comment]
> Should be fixed by #6670.

And zonkmachine taps out... :sunglasses:

**Comment by zonkmachine** [comment]
> Should be fixed by #6670.

I can confirm that this issue is fixed in #6670
